Task10_Auth/src/main/java/AuthSingleton.java

```diff
@@ -0,0 +1,114 @@
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.apache.http.HttpResponse;
+import org.apache.http.NameValuePair;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class AuthSingleton {
+    private static AuthSingleton object;
+    private static String write_token;
+    private static String read_token;
+    private static final String TOKEN_URI = "http://localhost:5353/oauth/token";
+    private static final String PASSWORD = "X7eBCXqlFC7x-mjxG5H91IRv_Bqe1oq7ZwXNA8aq";
+    private static final String USERNAME = "0oa157tvtugfFXEhU4x7";
+
+    private AuthSingleton() {
+        CredentialsProvider provider = new BasicCredentialsProvider();
+        UsernamePasswordCredentials credentials = new UsernamePasswordCredentials(USERNAME, PASSWORD);
+        provider.setCredentials(AuthScope.ANY, credentials);
+        HttpPost httpPost = new HttpPost(TOKEN_URI);
+        List<NameValuePair> params = new ArrayList();
+        params.add(new BasicNameValuePair("grant_type", "client_credentials"));
+        params.add(new BasicNameValuePair("scope", "read"));
+
+        try {
+            httpPost.setEntity(new UrlEncodedFormEntity(params));
+        } catch (UnsupportedEncodingException var20) {
+            var20.printStackTrace();
+        }
+
+        HttpClient client = HttpClientBuilder.create().setDefaultCredentialsProvider(provider).build();
+        HttpResponse response = null;
+
+        try {
+            response = client.execute(httpPost);
```

Comment:
You must close http response after execution. You will need to change HttpResponse type to CloseableHttpResponse

Reply:
HttpClient changed to CloseableHttpClient
HttpRespnse changed to ClosableHttpResponse
response.closed() added, were necessary